src/runtime/hexagon_dma_pool.h

```diff
@@ -0,0 +1,25 @@
+#ifndef _HEXAGON_DMA_POOL_H_
+#define _HEXAGON_DMA_POOL_H_
+
+namespace Halide { namespace Runtime { namespace Internal { namespace Hexagon {
+//Global Variables
+WEAK void *hexagon_cache_pool = NULL;
```

Comment:
I think this should be in the implementation too. It's only used in one place, to check if it is non-null before calling halide_hexagon_free_l2_pool, but that function could easily just not do anything if the pool is non-null (just like it is OK to delete a nullptr in C++).

Reply:
The problem here is with mock tests. Since we are not loading hexagon_cache_allocator module for target host , and the function definition of  function halide_hexagon_free_l2_pool is only in this module, there is a linking issue while running mock tests. If I move the hexagon_cache_pool to cpp file then the only option to run mocks is  to put a mock definition of halide_hexagon_free_l2_pool in mock_dma_implementation.cpp

Reply:
A mock definition sounds reasonable to me (better even), and it's worth keeping the implementation of the pool completely encapsulated if possible (no externally visible globals in the header).

Reply:
Agreed and made the changes.